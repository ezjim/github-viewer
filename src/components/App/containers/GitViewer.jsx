import React, { Component } from 'react';
import User from '../User/User';
import UserSearch from '../User/UserSearch';
import Repos from '../repo/Repos';
import { fetchUserInfo, fetchUserRepos } from '../repo/services/getRepos.jsx';


export default class GitViewer extends Component {
  state = {
    username: '',
    user: {
      avatar_url: '',
      login: '',
      location: '',
      following: 0,
      html_url: ''
    }
    // repos: []
    // search: false
  }

  handleUserChange = ({ target }) => {
    this.setState({ username: target.value });
  }

  handleUserSubmit = () => {
    fetchUserInfo(this.state.username);
    fetchUserRepos(this.state.username)
      .then(repos => this.setState({ repos, search: true }));
  }

  render() {
    let searched = '';
    if(this.state.search) {
      searched = 
      <>
        <User {...this.state.user} /> 
        <Repos repos={this.state.repos} />
      </>;
    }
    return (
      <>
        <UserSearch username={this.state.username} 
          onUserChange={this.handleUserChange} 
          onUserSubmit={this.handleUserSubmit}/>
        {searched}
      </>
    );
  }
}
