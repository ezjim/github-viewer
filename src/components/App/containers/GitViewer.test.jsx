import React from 'react';
import { shallow } from 'enzyme';
import GitViewer from './GitViewer';

describe('UserEnter component', () => {
  it('renders UserEnter', () => {
    const wrapper = shallow(<GitViewer 
      username="cficht"
      onUserChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
