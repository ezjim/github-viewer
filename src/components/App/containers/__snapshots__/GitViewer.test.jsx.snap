// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEnter component renders UserEnter 1`] = `
<Fragment>
  <UserSearch
    onUserChange={[Function]}
    onUserSubmit={[Function]}
    username=""
  />
</Fragment>
`;
